RELEASE.md

```diff
@@ -13,8 +13,14 @@ Publishing the helm chart is currently manual, and we should move to an automate
 - [ ] Run the `./hack/generate-changelog.sh` script to generate the a draft changelog
 - [ ] Update the CHANGELOG.md file with the generated changelog
 - [ ] Commit, Tag and Push to Gitlab
```

Comment:
It seems as if some fo the steps were lost somewhere. We no longer mention the steps to publish the image. Could we readd those?

Reply:
I don't follow, current https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/RELEASE.md RELEASE doesn't hold steps for image release. ALl we have is line 7
```md
Publishing the container is automated through gitlab-ci and only requires one to tag the commit and push it to gitlab.
```

Reply:
Sorry, I think I'm misremembering the steps we outlined in github issues / JIRA and the steps in this file. (see for example https://github.com/NVIDIA/cloud-native-team/issues/59).

The `Commit, Tag and Push` to Gitlab is the image trigger though, so we may want to reword this.

Not a blocker for this PR though.